sql/catalyst/src/main/scala/org/apache/spark/sql/Row.scala

```diff
@@ -335,6 +358,19 @@ trait Row extends Serializable {
   def getAs[T](fieldName: String): T = getAs[T](fieldIndex(fieldName))
 
   /**
+   * Returns the value of a given fieldName as a scala.util.Try object.
+   */
+  def attempt[T](fieldName: String): Try[T] = Try(getAs[T](fieldIndex(fieldName)))
```

Comment:
Those kind of problems will never manifest in an external Row with the wrong datatypes.  The execution engine will never produce values that are of a different type than the schema.